app/jobs/regular/user_email.rb

```diff
@@ -130,7 +131,7 @@ def message_for_email(user, post, type, notification, args = nil)
           return [nil, nil]
         end
 
-        unless user.user_option.email_always?
+        unless (user.user_option.email_always? && type != :user_private_message) || (user.user_option.email_private_messages_always? && type == :user_private_message)
```

Comment:
There is some code duplication between here and line 103. Perhaps we can extract the logic into a method?

Reply:
Thanks for raising this, there was a bug just above line 103 too, where `email_always` would affect all types of posts, including private messages. Fixed and added a test. 

(Will need to update tests to avoid stubs and fix merge conflict.)